scipy/stats/stats.py

```diff
@@ -7469,6 +7470,214 @@ def brunnermunzel(x, y, alternative="two-sided", distribution="t",
+                # lower-bound is the previous index
+                return k-1
+
+
+def confint_quantile(x, quantile, confidence, type='one-sided'):
+    r"""Computes non-parametric confidence intervals for any quantile
+    from a set of i.i.d. samples.
+
+    This function implements a non-parametric approach to compute
+    confidence intervals for quantiles. The approach is attributed to Thompson [1]_
+    and later proven to be applicable to any set of i.i.d. samples [2]_.
+    The computation is based on the observation that the probability of a quantile
+    :math:`q` to be larger than any sample :math:`x_m (1\leq m \leq N)`
+    can be computed as
+
+    .. math::
+
+        \mathbb{P}(x_m \leq q) = 1 - \sum_{k=0}^{m-1} \binom{N}{k} q^k(1-q)^{N-k}
+
+    Furthermore, these probabilities are symmetric, which allows to compute both
+    upper and lower bounds from the same computation:
+
+    .. math::
+
+        \mathbb{P}(x_m \leq q) = \mathbb{P}(x_{N-m+1} \geq 1-q).
+
+    The function computes confidence intervals for a given quantile and
+    confidence level, based on `x` which is either a set of samples
+    (one-dimensional array_like) or the number of samples available.
+    The confidence intervals are valid if and only if the samples are i.i.d.
+
+    Both one-sided and two-sided confidence intervals can be obtained
+    (default is one-sided). The function returns two values: either the bounds for the two one-sided
+    confidence intervals, or the lower and upper bounds of a two-sided confidence interval.
+    The return values are either the indexes of the bounds (if `x` is an integer) or
+    sample values (if `x` is the set of samples).
+    `None` is returned when there are not enough samples to compute
+    the desired confidence interval.
+
+    There is no uniqueness of the two-sided confidence interval (see Notes below).
+    Without further assumption on the samples (eg, the nature of the underlying distribution),
+    the one-sided intervals are optimally tight.
+
+    Parameters
+    ----------
+    x : array_like or int
+        Array of samples, should be one-dimensional.
+        If integer, taken as the number of samples available (strictly positive)
+    quantile : float
+        The quantile for which we want to compute the confidence interval.
+        Must be strictly between 0 and 1.
+    confidence : float
+        The desired confidence level of the confidence interval.
+        Must be strictly between 0 and 1.
+    type : {'one-sided', 'two-sided'}, optional
+        Defines the type of confidence interval computed.
+        Default is 'one-sided'.
+
+          * 'one-sided' : computes the best possible one-sided confidence intervals (both lower and upper bounds) for the given quantile.
+          * 'two-sided' : computes a two-sided confidence interval by combination of two one-sided intervals. E.g., a 90% two-sided interval is computed by combining two 95% one-sided intervals
```

Comment:
Are these lines a bit long maybe? Perhaps not a big deal

Reply:
This is super weird, when I break these into multiple lines, the compilation of the doc crashes... I tried many different formats, and eventually left it like that, as its the only way I found to get the doc to compile properly..?!